cdap-common/src/main/java/io/cdap/cdap/common/service/Retries.java

```diff
@@ -61,6 +62,17 @@ private Retries() {
     void run() throws T;
   }
 
+  /**
+   * A Callable whose throwable is generic and call method takes attempt count as argument.
+   * This is helpful for capturing retry count
+   *
+   * @param <V> the type of return value
+   * @param <T> the type of throwable
+   */
+  public interface CallableWithAttempt<V, T extends Throwable> {
+    V call(int attempt) throws T;
```

Comment:
Since we are introducing a new interface, I think it is better to have it as `V call(RetryContext context) throw T`, with the retry attempt accessible via the context object. (e.g. `int getRetryAttempt()`)

Reply:
Yes, that makes it easier to add more details later. Done.